etcdserver/auth/auth.go

```diff
@@ -281,11 +285,14 @@ func (s *store) UpdateUser(user User) (User, error) {
 		return old, err
 	}
 
-	hash, err := s.HashPassword(user.Password)
-	if err != nil {
-		return old, err
+	if user.Password != "" {
+		var hash string
+		hash, err = s.HashPassword(user.Password)
```

Comment:
Old password gets overwritten with the hash value of empty string if `user` was passed to `UpdateUser` with empty password.

The line here https://github.com/coreos/etcd/blob/master/etcdserver/auth/auth.go#L457-L459